src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -1892,6 +1913,10 @@ private static bool IsTypeSpecForTypicalInstantiation(TypeDesc t)
 
                     exactType = constrainedType;
                 }
+                else if (isStaticVirtual)
+                {
+                    pResult->thisTransform = CORINFO_THIS_TRANSFORM.CORINFO_NO_THIS_TRANSFORM;
+                }
```

Comment:
Is it actually possible to reach this code? My reading indicates that either the directMethod will be resolved to a non-null value above, or we will throw a RequiresRuntimeJitException.

Reply:
I have reformulated this bit and I believe it's now reachable.